### PR TITLE
initialize project dependencies

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* Update gitignore
* Add pom dependency
* Add config for style checking and bugs finding

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+                <artifactId>cobertura-maven-plugin</artifactId>
+                <version>2.7</version>
+                <configuration>
+                    <formats>
+                        <format>html</format>
+                        <format>xml</format>
+                    </formats>
+                    <check>
+                        <haltOnFailure>true</haltOnFailure>
+                        <branchRate>65</branchRate>
```
Comment (yungezz) [inline pom.xml]: you might want to disable bar check for now, otherwise build will fail if coverage bar doesn't meet

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+                        <totalBranchRate>65</totalBranchRate>
+                    </check>
+                    <instrumentation>
+                        <excludes>
+                            <exclude>com/microsoft/azure/**/GetHashMac.class</exclude>
```
Comment (yungezz) [inline pom.xml]: file path aligns with package name, there's no 'azure' in package name

Comment (Incarnation-p-lee) [reply pom.xml]: Oh, miss this old package, may no excludes for now.

File: pom.xml (hunk 1 above)
Comment (Incarnation-p-lee) [reply pom.xml]: yes, copy that.

**Review by yungezz — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -20,3 +20,6 @@
 
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
+
+.idea*
```
Comment (yungezz) [inline .gitignore:24]: you may want to copy .gitingore from previous repo, which contains more, eg. eclipse files .,

**Pull request merged**